### PR TITLE
Mlh0079 2218

### PR DESCRIPTION
**Pull request by Michaellh0079**

### CODE REVIEW

File: task/main.py [hunk 1]
```diff
@@ -132,47 +121,26 @@ def get_file_links_http(self, url_path, file_reg_ex=None, dir_reg_ex=None, depth
...
-                                             dir_reg_ex=dir_reg_ex, depth=(depth - 1))
+            fetched_html = self.html_request(url_path)
+            directory_list = []
+            for a_tag in fetched_html.findAll('a', href=True):
+                url_segment = a_tag['href'].rstrip('/').rsplit('/', 1)[-1]
+                path = f"{url_path}{url_segment}"
+                if '.' in url_segment:
+                    head_resp = self.session.head(path).headers
+                    granule_dict[path] = {}
+                    granule_dict[path]['ETag'] = head_resp['ETag']
```
Comment (amarouane-ABDELHAK) [inline task/main.py:132]: Consider the case where Etag is not present.

File: task/main.py [hunk 2]
```diff
@@ -132,47 +121,26 @@ def get_file_links_http(self, url_path, file_reg_ex=None, dir_reg_ex=None, depth
...
+                    granule_dict[path]['Last-Modified'] = head_resp['Last-Modified'].replace(',', '').replace(' ', '')
```
Comment (amarouane-ABDELHAK) [inline task/main.py:133]: Use python parser to parse the date as provided by the server.

File: task/main.py [hunk 3]
```diff
@@ -132,47 +121,26 @@ def get_file_links_http(self, url_path, file_reg_ex=None, dir_reg_ex=None, depth
...
-            for directory in discovered_directories:
-                granule_dict.update(
-                    self.get_file_links_http(url_path=directory, file_reg_ex=file_reg_ex,
-                                             dir_reg_ex=dir_reg_ex, depth=(depth - 1))
+            fetched_html = self.html_request(url_path)
+            directory_list = []
+            for a_tag in fetched_html.findAll('a', href=True):
+                url_segment = a_tag['href'].rstrip('/').rsplit('/', 1)[-1]
+                path = f"{url_path}{url_segment}"
+                if '.' in url_segment:
```
Comment (amarouane-ABDELHAK) [inline task/main.py:129]: Add a comment to explain why we check for '.'

File: task/main.py [hunk 4]
```diff
@@ -132,47 +121,26 @@ def get_file_links_http(self, url_path, file_reg_ex=None, dir_reg_ex=None, depth
...
+                    head_resp = self.session.head(path).headers
```
Comment (amarouane-ABDELHAK) [inline task/main.py:130]: Move this snippet to a different function.

**Comment by Michaellh0079** [comment]
This merge is no longer needed as everything will be resolved in the 2277 branch.

**Pull request closed without merging**